src/main/java/frc/robot/commands/RunPath.java

```diff
@@ -46,36 +50,39 @@ public RunPath(List leftPath, List rightPath) {
   // Called when the command is initially scheduled.
   @Override
   public void initialize() {
+    m_leftIterator = leftPath.iterator();
+    m_rightIterator = rightPath.iterator();
     i = 0;
     timer.reset();
     timer.start();
   }
 
   // Called every time the scheduler runs while the command is scheduled.
   @Override
-  public void execute() {
-    
-    RobotContainer.mDriveSubsystem.setLeftPidVelocitySetpoint(-1*RobotContainer.mDriveSubsystem.fpsToRPM(Double.valueOf(leftPath.get(i).toString())));
-    RobotContainer.mDriveSubsystem.setRightPidVelocitySetpoint(RobotContainer.mDriveSubsystem.fpsToRPM(Double.valueOf(rightPath.get(i).toString())));
-    SmartDashboard.putNumber("Left Commanded Velocity", RobotContainer.mDriveSubsystem.fpsToRPM(Double.valueOf(leftPath.get(i).toString())));    
-    SmartDashboard.putNumber("Right Commanded Velocity", RobotContainer.mDriveSubsystem.fpsToRPM(Double.valueOf(rightPath.get(i).toString())));
-
-    i++;
+  public void execute(){
+    if(i< leftPath.size()){
+      RobotContainer.mDriveSubsystem.setLeftPidVelocitySetpoint(-1*RobotContainer.mDriveSubsystem.fpsToRPM(m_leftIterator.next()));
+      RobotContainer.mDriveSubsystem.setRightPidVelocitySetpoint(RobotContainer.mDriveSubsystem.fpsToRPM(m_rightIterator.next()));
+      SmartDashboard.putNumber("Left Commanded Velocity", -1*RobotContainer.mDriveSubsystem.fpsToRPM(Double.valueOf(leftPath.get(i).toString())));    
+      SmartDashboard.putNumber("Right Commanded Velocity", RobotContainer.mDriveSubsystem.fpsToRPM(Double.valueOf(rightPath.get(i).toString())));
+      i++;
+    }
+  
   }
 
   // Called once the command ends or is interrupted.
   @Override
   public void end(boolean interrupted) {
+    m_rightIterator.remove();
+    m_leftIterator.remove();
     RobotContainer.mDriveSubsystem.setLeftPidVelocitySetpoint(0);
```

Comment:
This should probably be refactored, as it will use PID to attempt to brake the robot, the end case might want to be coast to break. so sparkMax.set(0) to just set the command of the motor to 0 percent.